third_party/amd/backend/compiler.py

```diff
@@ -258,6 +258,7 @@ def make_llir(src, metadata, options):
             paths = [path for (name, path) in options.extern_libs if amd.need_extern_lib(llvm_mod, name)]
             llvm.link_extern_libs(llvm_mod, paths)
 
+        llvm.attach_datalayout(llvm_mod, amd.TARGET_TRIPLE, options.arch, '')
```

Comment:
Do we want to attach right after we get the llvm module at L237 like the target triple?